docs/write_test.rst

```diff
@@ -164,20 +164,20 @@ Use a comma-separated list for multiple input values:
     import pytest
     import math
 
-    @pytest.mark.parametrize('x,y', [(1, 2), (1, 3), (2, 1)])
+    @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
     @ti.test()
-    def test_atan2(x, y):
+    def test_atan2(y, x):
         r = ti.field(ti.f32, ())
         s = ti.field(ti.f32, ())
 
         @ti.kernel
         def foo():
-            r[None] = ti.atan2(r[None])
+            r[None] = ti.atan2(r[None], s[None])
 
         r[None] = x
         s[None] = y
```

Comment:
I wonder if the `x` and `y` here need to be swapped too?

Reply:
Yes, u are right. I think I should double check this file.

Reply:
Actually this script  cannot pass the test, due to line 5. 
```python
import taichi as ti
import pytest
import math
@pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])#fixture 'taichi_archs' not found
@ti.test()
def test_atan2(y, x):
    r = ti.field(ti.f32, ())
    s = ti.field(ti.f32, ())
    print(x)
    print(y)
    @ti.kernel
    def foo():
        r[None] = ti.atan2(r[None], s[None])
    r[None] = y
    s[None] = x
    foo()
    assert r[None] == math.atan2(y, x)
```
Then I modify it to the  second script and the terminal show that r[None] and math.atan2(y, x) are not the same.

```python
import taichi as ti
    import pytest
    import math

    @pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
    @ti.test()
    def test_atan2(y, x):
        r = ti.field(ti.f32, ())
        s = ti.field(ti.f32, ())

        @ti.kernel
        def foo():
            r[None] = ti.atan2(r[None], s[None])

        r[None] = y
        s[None] = x
        foo()
        assert r[None] == math.atan2(y, x)#not the same!!!
```
I modify it to the  third script and finally this script passes the test.
```python
import taichi as ti
import pytest
import math
@ti.all_archs
@pytest.mark.parametrize('y,x', [(1, 2), (1, 3), (2, 1)])
def test_atan2(y, x):
    r = ti.field(ti.f32, ())
    s = ti.field(ti.f32, ())
    print(x)
    print(y)
    @ti.kernel
    def foo():
        r[None] = ti.atan2(r[None], s[None])
    r[None] = y
    s[None] = x
    foo()
    assert ti.approx(r[None]) == math.atan2(y, x)
```

Reply:
> Then I modify it to the second script and the terminal show that r[None] and math.atan2(y, x) are not the same.

Interesting! I also found that they're not the same:
```
FAILED test2.py::test_atan2[2-1] - assert 1.1071487665176392 == 1.1071487177940904
```
But I can't run the script either. I modified `@ti.test()` in the second script to `@ti.all_archs` to run it.

@archibate Do you have any idea on `@ti.test()` itself runs successfully but `@pytest.mark.parametrize` together with `@ti.test` fails?